lib/ansible/plugins/action/__init__.py

```diff
@@ -909,6 +910,12 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
         if data.pop("_ansible_suppress_tmpdir_delete", False):
             self._cleanup_remote_tmp = False
 
+        # NOTE: yum returns results .. but that made it 'compatible' with squashing, so we allow mappings, for now
+        if 'results' in data and not isinstance(data['results'], Mapping):
```

Comment:
IIRC `results` in the `yum` module is a `list`.

Reply:
actually, `results` should always be a list .. of dicts, so this was wrong